test/with_dummyserver/test_proxy_poolmanager.py

```diff
@@ -22,343 +22,332 @@ def setUp(self):
         self.proxy_url = "http://%s:%d" % (self.proxy_host, self.proxy_port)
 
     def test_basic_proxy(self):
-        http = proxy_from_url(self.proxy_url, ca_certs=DEFAULT_CA)
-        self.addCleanup(http.clear)
+        with proxy_from_url(self.proxy_url, ca_certs=DEFAULT_CA) as http:
```

Comment:
Should we change `proxy_from_url` to just `ProxyManager` calls ?

Reply:
Sure, but not as part of this PR (because yours did not that either, probaby because it would have been a breaking change). I added it to my list in #114 so I won't forget it.

Reply:
Then I think it's ready to to be merged.